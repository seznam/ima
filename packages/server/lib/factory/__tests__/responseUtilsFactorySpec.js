import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';

const responseUtilsFactory = require('../responseUtilsFactory.js');

const manifestMock = require('../__mocks__/manifest.json');

vi.mock('crypto', () => ({
  randomUUID: () => 'UUID',
}));

vi.mock('fs', () => {
  const { toMockedInstance } = vi.requireActual('to-mock');
  const originalModule = vi.requireActual('fs');

  return {
    ...toMockedInstance(originalModule, {
      existsSync() {
        return true;
      },
      readFileSync(path) {
        if (path.endsWith('manifest.json')) {
          return JSON.stringify(manifestMock);
        }

        return 'runner#{scriptResources}';
      },
    }),
  };
});

describe('responseUtilsFactory', () => {
  const nowMock = vi.spyOn(global.Date, 'now').mockReturnValue('now');
  const randomMock = vi.spyOn(global.Math, 'random').mockReturnValue(0);

  const {
    processContent,
    createContentVariables,
    sendResponseHeaders,
    _prepareCookieOptionsForExpress,
  } = responseUtilsFactory({ applicationFolder: 'applicationFolder' });

  nowMock.mockRestore();
  randomMock.mockRestore();

  let event;

  beforeEach(() => {
    event = {
      res: {
        cookie: vi.fn(),
        set: vi.fn(),
        locals: {},
      },
      context: {
        response: {
          content: '<html>#{styles}#{revivalSettings}#{runner}</html>',
          contentVariables: {},
          page: {},
        },
        bootConfig: {
          settings: {
            $Language: 'en',
            $Debug: true,
          },
        },
      },
    };
  });

  afterEach(() => {
    vi.resetAllMocks();
  });

  describe('_prepareCookieOptionsForExpress', () => {
    it('should convert cookie maxAge to ms for Express', () => {
      let options = { maxAge: 1 };
      let expressOptions = _prepareCookieOptionsForExpress(options);

      expect(options.maxAge).toBe(1);
      expect(expressOptions.maxAge).toBe(1000);
    });

    it('should remove cookie maxAge: null for Express', () => {
      // Because Express converts null to 0, which is not intended.
      let options = { maxAge: null };
      let expressOptions = _prepareCookieOptionsForExpress(options);

      expect(options.maxAge).toBeNull();
      expect(expressOptions.maxAge).toBeUndefined();
    });
  });

  describe('sendResponseHeaders', () => {
    it('should set page cookie to response headers', () => {
      const cookie = new Map();
      cookie.set('namex', { value: 1, options: {} });
      cookie.set('namey', { value: 2, options: {} });

      event.context.response.page.cookie = cookie;

      sendResponseHeaders(event);

      expect(event.res.cookie).toHaveBeenCalledTimes(2);
      expect(event.res.cookie.mock.calls[0]).toEqual(['namex', 1, {}]);
      expect(event.res.cookie.mock.calls[1]).toEqual(['namey', 2, {}]);
    });

    it('should set page headers to response headers', () => {
      event.context.response.page.headers = {
        'Content-Type': 'text/html',
      };

      sendResponseHeaders(event);

      expect(event.res.set).toHaveBeenCalledTimes(1);
      expect(event.res.set.mock.calls[0][0]).toEqual(
        event.context.response.page.headers
      );
    });
  });
  describe('createContentVariables', () => {
    it('should return empty object if there is no valid bootConfig', () => {
      expect(createContentVariables({ context: {} })).toStrictEqual({});
      expect(
        createContentVariables({ context: { bootConfig: null } })
      ).toStrictEqual({});
      expect(
        createContentVariables({ context: { bootConfig: {} } })
      ).toStrictEqual({});
      expect(
        createContentVariables({ context: { bootConfig: { settings: null } } })
      ).toStrictEqual({});
    });

    it('should generate base set of content variables', () => {
      expect(createContentVariables(event)).toMatchSnapshot();
    });
  });

  describe('processContent', () => {
    it('should return original content without any boot config', () => {
      event.context.response.content = 'content';

      expect(processContent(event)).toBe('content');
    });

    it('should interpolate revival scripts into page content', () => {
      event.context.response.contentVariables = createContentVariables(event);

      const content = processContent(event);
      expect(content).toMatchSnapshot();
    });

    it('should allow overrides through custom $Source definition', () => {
      event.context.bootConfig.settings.$Resources = (
        context,
        manifest,
        sources
      ) => {
        return {
          styles: [],
          esScripts: [...sources.scripts, 'custom-script-src'],
        };
      };
      event.context.response.contentVariables = createContentVariables(event);

      const content = processContent(event);

      expect(content).toMatchSnapshot();
    });

    it('should interpolate deep embedded variables', () => {
      event.context.response.content = '<html>#{v1}</html>';
      event.context.response.contentVariables = {
        v1: '#{v2}',
        v2: '#{v3}',
        v3: '#{v4}',
        v4: '#{v5}',
        v5: '#{v6}',
        v6: '#{v7}',
        v7: 'final v7 content',
      };

      const content = processContent(event);

      expect(content).toBe('<html>final v7 content</html>');
    });

    it('should allow interpolation of settings variables from bootConfig', () => {
      event.context.bootConfig.settings.$Version = '1.0.0';
      event.context.response.content = '<html>#{c1}: #{$Version}</html>';
      event.context.response.contentVariables = {
        c1: 'content-variable',
      };

      const content = processContent(event);

      expect(content).toBe('<html>content-variable: 1.0.0</html>');
    });
  });
});
