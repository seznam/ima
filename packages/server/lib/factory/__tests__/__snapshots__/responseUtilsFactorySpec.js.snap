// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`responseUtilsFactory > createContentVariables > should generate base set of content variables 1`] = `
{
  "_": {
    "manifest": {},
    "resources": {
      "esScripts": [],
      "esStyles": [],
      "scripts": [],
      "styles": [],
    },
  },
  "meta": "",
  "revivalCache": "<script id="ima-revival-cache">(function (root) {
      root.$IMA = root.$IMA || {};
      $IMA.Cache = {};
    })(typeof window !== 'undefined' && window !== null ? window : global);
    </script>",
  "revivalSettings": "<script id="ima-revival-settings">(function (root) {
      root.$Debug = true;
      root.$IMA = root.$IMA || {};
      $IMA.SPA = false;
      $IMA.$PublicPath = "";
      $IMA.$RequestID = "now-0-12f4c553-a5f3-4545-ba03-ff938fb8f943";
      $IMA.$Language = "en";
      $IMA.$Env = "undefined";
      $IMA.$Debug = true;
      $IMA.$Version = "undefined";
      $IMA.$App = undefined;
      $IMA.$Protocol = "undefined";
      $IMA.$Host = "undefined";
      $IMA.$Path = "undefined";
      $IMA.$Root = "undefined";
      $IMA.$LanguagePartPath = "undefined";
    })(typeof window !== 'undefined' && window !== null ? window : global);
    </script>",
  "runner": "<script id="ima-runner"></script>",
  "scriptResources": "{\\"scripts\\":[],\\"esScripts\\":[]}",
  "styles": "",
}
`;

exports[`responseUtilsFactory > processContent > should allow overrides through custom $Source definition 1`] = `
"<html><script id="ima-revival-settings">(function (root) {
      root.$Debug = true;
      root.$IMA = root.$IMA || {};
      $IMA.SPA = false;
      $IMA.$PublicPath = "";
      $IMA.$RequestID = "now-0-b8aecc4d-e429-4bba-a42f-b3ec11a57703";
      $IMA.$Language = "en";
      $IMA.$Env = "undefined";
      $IMA.$Debug = true;
      $IMA.$Version = "undefined";
      $IMA.$App = undefined;
      $IMA.$Protocol = "undefined";
      $IMA.$Host = "undefined";
      $IMA.$Path = "undefined";
      $IMA.$Root = "undefined";
      $IMA.$LanguagePartPath = "undefined";
    })(typeof window !== 'undefined' && window !== null ? window : global);
    </script><script id="ima-runner"></script></html>"
`;

exports[`responseUtilsFactory > processContent > should interpolate revival scripts into page content 1`] = `
"<html><script id="ima-revival-settings">(function (root) {
      root.$Debug = true;
      root.$IMA = root.$IMA || {};
      $IMA.SPA = false;
      $IMA.$PublicPath = "";
      $IMA.$RequestID = "now-0-6e45a64b-29c1-4bad-8c1a-9a53987c8b88";
      $IMA.$Language = "en";
      $IMA.$Env = "undefined";
      $IMA.$Debug = true;
      $IMA.$Version = "undefined";
      $IMA.$App = undefined;
      $IMA.$Protocol = "undefined";
      $IMA.$Host = "undefined";
      $IMA.$Path = "undefined";
      $IMA.$Root = "undefined";
      $IMA.$LanguagePartPath = "undefined";
    })(typeof window !== 'undefined' && window !== null ? window : global);
    </script><script id="ima-runner"></script></html>"
`;

exports[`responseUtilsFactory createContentVariables should generate base set of content variables 1`] = `
{
  "_": {
    "manifest": {
      "assets": {
        "server/app.server.js": {
          "contenthash": "6e0337810e75fd13",
          "fileName": "server/app.server.6e0337810e75fd13.js",
          "immutable": true,
          "javascriptModule": false,
          "name": "server/app.server.js",
          "related": {
            "sourceMap": "server/app.server.6e0337810e75fd13.js.map",
          },
        },
        "server/locale/cs.js": {
          "contenthash": "924fe4e0a21c1e61",
          "fileName": "server/locale/cs.924fe4e0a21c1e61.js",
          "immutable": true,
          "javascriptModule": false,
          "name": "server/locale/cs.js",
          "related": {
            "sourceMap": "server/locale/cs.924fe4e0a21c1e61.js.map",
          },
        },
        "server/locale/en.js": {
          "contenthash": "10dc65d1067ddbb6",
          "fileName": "server/locale/en.10dc65d1067ddbb6.js",
          "immutable": true,
          "javascriptModule": false,
          "name": "server/locale/en.js",
          "related": {
            "sourceMap": "server/locale/en.10dc65d1067ddbb6.js.map",
          },
        },
        "server/vendors.js": {
          "contenthash": "aa3dff21bb6fa241",
          "fileName": "server/vendors.aa3dff21bb6fa241.js",
          "immutable": true,
          "javascriptModule": false,
          "name": "server/vendors.js",
          "related": {
            "sourceMap": "server/vendors.aa3dff21bb6fa241.js.map",
          },
        },
        "static/css/app.css": {
          "contenthash": "d0ad44d05f82db5f",
          "fileName": "static/css/app.d0ad44d05f82db5f.css",
          "immutable": true,
          "minimized": true,
          "name": "static/css/app.css",
          "related": {
            "brotliCompressed": "static/css/app.d0ad44d05f82db5f.css.br",
            "gzipped": "static/css/app.d0ad44d05f82db5f.css.gz",
            "sourceMap": "static/css/app.d0ad44d05f82db5f.css.map",
          },
        },
        "static/js.es/app.client.js": {
          "contenthash": "2106a34c6b8bbad8",
          "fileName": "static/js.es/app.client.2106a34c6b8bbad8.js",
          "immutable": true,
          "javascriptModule": false,
          "minimized": true,
          "name": "static/js.es/app.client.js",
          "related": {
            "brotliCompressed": "static/js.es/app.client.2106a34c6b8bbad8.js.br",
            "gzipped": "static/js.es/app.client.2106a34c6b8bbad8.js.gz",
            "sourceMap": "static/js.es/app.client.2106a34c6b8bbad8.js.map",
          },
        },
        "static/js.es/locale/cs.js": {
          "contenthash": "c1b3211af4424f2c",
          "fileName": "static/js.es/locale/cs.c1b3211af4424f2c.js",
          "immutable": true,
          "javascriptModule": false,
          "minimized": true,
          "name": "static/js.es/locale/cs.js",
          "related": {
            "brotliCompressed": "static/js.es/locale/cs.c1b3211af4424f2c.js.br",
            "gzipped": "static/js.es/locale/cs.c1b3211af4424f2c.js.gz",
            "sourceMap": "static/js.es/locale/cs.c1b3211af4424f2c.js.map",
          },
        },
        "static/js.es/locale/en.js": {
          "contenthash": "371127fdbfbe93d2",
          "fileName": "static/js.es/locale/en.371127fdbfbe93d2.js",
          "immutable": true,
          "javascriptModule": false,
          "minimized": true,
          "name": "static/js.es/locale/en.js",
          "related": {
            "brotliCompressed": "static/js.es/locale/en.371127fdbfbe93d2.js.br",
            "gzipped": "static/js.es/locale/en.371127fdbfbe93d2.js.gz",
            "sourceMap": "static/js.es/locale/en.371127fdbfbe93d2.js.map",
          },
        },
        "static/js.es/vendors.js": {
          "contenthash": "a873907f25297544",
          "fileName": "static/js.es/vendors.a873907f25297544.js",
          "immutable": true,
          "javascriptModule": false,
          "minimized": true,
          "name": "static/js.es/vendors.js",
          "related": {
            "brotliCompressed": "static/js.es/vendors.a873907f25297544.js.br",
            "gzipped": "static/js.es/vendors.a873907f25297544.js.gz",
            "sourceMap": "static/js.es/vendors.a873907f25297544.js.map",
          },
        },
        "static/js/app.client.js": {
          "contenthash": "8dc14fd2c9e52eef",
          "fileName": "static/js/app.client.8dc14fd2c9e52eef.js",
          "immutable": true,
          "javascriptModule": false,
          "minimized": true,
          "name": "static/js/app.client.js",
          "related": {
            "brotliCompressed": "static/js/app.client.8dc14fd2c9e52eef.js.br",
            "gzipped": "static/js/app.client.8dc14fd2c9e52eef.js.gz",
            "sourceMap": "static/js/app.client.8dc14fd2c9e52eef.js.map",
          },
        },
        "static/js/locale/cs.js": {
          "contenthash": "2547d226f675e927",
          "fileName": "static/js/locale/cs.2547d226f675e927.js",
          "immutable": true,
          "javascriptModule": false,
          "minimized": true,
          "name": "static/js/locale/cs.js",
          "related": {
            "brotliCompressed": "static/js/locale/cs.2547d226f675e927.js.br",
            "gzipped": "static/js/locale/cs.2547d226f675e927.js.gz",
            "sourceMap": "static/js/locale/cs.2547d226f675e927.js.map",
          },
        },
        "static/js/locale/en.js": {
          "contenthash": "3c6a5e7a55bb2ab4",
          "fileName": "static/js/locale/en.3c6a5e7a55bb2ab4.js",
          "immutable": true,
          "javascriptModule": false,
          "minimized": true,
          "name": "static/js/locale/en.js",
          "related": {
            "brotliCompressed": "static/js/locale/en.3c6a5e7a55bb2ab4.js.br",
            "gzipped": "static/js/locale/en.3c6a5e7a55bb2ab4.js.gz",
            "sourceMap": "static/js/locale/en.3c6a5e7a55bb2ab4.js.map",
          },
        },
        "static/js/vendors.js": {
          "contenthash": "0e456297851f0a3a",
          "fileName": "static/js/vendors.0e456297851f0a3a.js",
          "immutable": true,
          "javascriptModule": false,
          "minimized": true,
          "name": "static/js/vendors.js",
          "related": {
            "brotliCompressed": "static/js/vendors.0e456297851f0a3a.js.br",
            "gzipped": "static/js/vendors.0e456297851f0a3a.js.gz",
            "sourceMap": "static/js/vendors.0e456297851f0a3a.js.map",
          },
        },
      },
      "assetsByCompiler": {
        "client": {
          "static/js/app.client.js": {
            "contenthash": "8dc14fd2c9e52eef",
            "fileName": "static/js/app.client.8dc14fd2c9e52eef.js",
            "immutable": true,
            "javascriptModule": false,
            "minimized": true,
            "name": "static/js/app.client.js",
            "related": {
              "brotliCompressed": "static/js/app.client.8dc14fd2c9e52eef.js.br",
              "gzipped": "static/js/app.client.8dc14fd2c9e52eef.js.gz",
              "sourceMap": "static/js/app.client.8dc14fd2c9e52eef.js.map",
            },
          },
          "static/js/locale/cs.js": {
            "contenthash": "2547d226f675e927",
            "fileName": "static/js/locale/cs.2547d226f675e927.js",
            "immutable": true,
            "javascriptModule": false,
            "minimized": true,
            "name": "static/js/locale/cs.js",
            "related": {
              "brotliCompressed": "static/js/locale/cs.2547d226f675e927.js.br",
              "gzipped": "static/js/locale/cs.2547d226f675e927.js.gz",
              "sourceMap": "static/js/locale/cs.2547d226f675e927.js.map",
            },
          },
          "static/js/locale/en.js": {
            "contenthash": "3c6a5e7a55bb2ab4",
            "fileName": "static/js/locale/en.3c6a5e7a55bb2ab4.js",
            "immutable": true,
            "javascriptModule": false,
            "minimized": true,
            "name": "static/js/locale/en.js",
            "related": {
              "brotliCompressed": "static/js/locale/en.3c6a5e7a55bb2ab4.js.br",
              "gzipped": "static/js/locale/en.3c6a5e7a55bb2ab4.js.gz",
              "sourceMap": "static/js/locale/en.3c6a5e7a55bb2ab4.js.map",
            },
          },
          "static/js/vendors.js": {
            "contenthash": "0e456297851f0a3a",
            "fileName": "static/js/vendors.0e456297851f0a3a.js",
            "immutable": true,
            "javascriptModule": false,
            "minimized": true,
            "name": "static/js/vendors.js",
            "related": {
              "brotliCompressed": "static/js/vendors.0e456297851f0a3a.js.br",
              "gzipped": "static/js/vendors.0e456297851f0a3a.js.gz",
              "sourceMap": "static/js/vendors.0e456297851f0a3a.js.map",
            },
          },
        },
        "client.es": {
          "static/css/app.css": {
            "contenthash": "d0ad44d05f82db5f",
            "fileName": "static/css/app.d0ad44d05f82db5f.css",
            "immutable": true,
            "minimized": true,
            "name": "static/css/app.css",
            "related": {
              "brotliCompressed": "static/css/app.d0ad44d05f82db5f.css.br",
              "gzipped": "static/css/app.d0ad44d05f82db5f.css.gz",
              "sourceMap": "static/css/app.d0ad44d05f82db5f.css.map",
            },
          },
          "static/js.es/app.client.js": {
            "contenthash": "2106a34c6b8bbad8",
            "fileName": "static/js.es/app.client.2106a34c6b8bbad8.js",
            "immutable": true,
            "javascriptModule": false,
            "minimized": true,
            "name": "static/js.es/app.client.js",
            "related": {
              "brotliCompressed": "static/js.es/app.client.2106a34c6b8bbad8.js.br",
              "gzipped": "static/js.es/app.client.2106a34c6b8bbad8.js.gz",
              "sourceMap": "static/js.es/app.client.2106a34c6b8bbad8.js.map",
            },
          },
          "static/js.es/locale/cs.js": {
            "contenthash": "c1b3211af4424f2c",
            "fileName": "static/js.es/locale/cs.c1b3211af4424f2c.js",
            "immutable": true,
            "javascriptModule": false,
            "minimized": true,
            "name": "static/js.es/locale/cs.js",
            "related": {
              "brotliCompressed": "static/js.es/locale/cs.c1b3211af4424f2c.js.br",
              "gzipped": "static/js.es/locale/cs.c1b3211af4424f2c.js.gz",
              "sourceMap": "static/js.es/locale/cs.c1b3211af4424f2c.js.map",
            },
          },
          "static/js.es/locale/en.js": {
            "contenthash": "371127fdbfbe93d2",
            "fileName": "static/js.es/locale/en.371127fdbfbe93d2.js",
            "immutable": true,
            "javascriptModule": false,
            "minimized": true,
            "name": "static/js.es/locale/en.js",
            "related": {
              "brotliCompressed": "static/js.es/locale/en.371127fdbfbe93d2.js.br",
              "gzipped": "static/js.es/locale/en.371127fdbfbe93d2.js.gz",
              "sourceMap": "static/js.es/locale/en.371127fdbfbe93d2.js.map",
            },
          },
          "static/js.es/vendors.js": {
            "contenthash": "a873907f25297544",
            "fileName": "static/js.es/vendors.a873907f25297544.js",
            "immutable": true,
            "javascriptModule": false,
            "minimized": true,
            "name": "static/js.es/vendors.js",
            "related": {
              "brotliCompressed": "static/js.es/vendors.a873907f25297544.js.br",
              "gzipped": "static/js.es/vendors.a873907f25297544.js.gz",
              "sourceMap": "static/js.es/vendors.a873907f25297544.js.map",
            },
          },
        },
        "server": {
          "server/app.server.js": {
            "contenthash": "6e0337810e75fd13",
            "fileName": "server/app.server.6e0337810e75fd13.js",
            "immutable": true,
            "javascriptModule": false,
            "name": "server/app.server.js",
            "related": {
              "sourceMap": "server/app.server.6e0337810e75fd13.js.map",
            },
          },
          "server/locale/cs.js": {
            "contenthash": "924fe4e0a21c1e61",
            "fileName": "server/locale/cs.924fe4e0a21c1e61.js",
            "immutable": true,
            "javascriptModule": false,
            "name": "server/locale/cs.js",
            "related": {
              "sourceMap": "server/locale/cs.924fe4e0a21c1e61.js.map",
            },
          },
          "server/locale/en.js": {
            "contenthash": "10dc65d1067ddbb6",
            "fileName": "server/locale/en.10dc65d1067ddbb6.js",
            "immutable": true,
            "javascriptModule": false,
            "name": "server/locale/en.js",
            "related": {
              "sourceMap": "server/locale/en.10dc65d1067ddbb6.js.map",
            },
          },
          "server/vendors.js": {
            "contenthash": "aa3dff21bb6fa241",
            "fileName": "server/vendors.aa3dff21bb6fa241.js",
            "immutable": true,
            "javascriptModule": false,
            "name": "server/vendors.js",
            "related": {
              "sourceMap": "server/vendors.aa3dff21bb6fa241.js.map",
            },
          },
        },
      },
      "publicPath": "/",
    },
    "resources": {
      "esScripts": [
        [
          "/static/js.es/app.client.2106a34c6b8bbad8.js",
          {
            "async": "",
            "crossorigin": "anonymous",
          },
        ],
        [
          "/static/js.es/locale/en.371127fdbfbe93d2.js",
          {
            "async": "",
            "crossorigin": "anonymous",
          },
        ],
        [
          "/static/js.es/vendors.a873907f25297544.js",
          {
            "async": "",
            "crossorigin": "anonymous",
          },
        ],
      ],
      "esStyles": [
        [
          "/static/css/app.d0ad44d05f82db5f.css",
          {
            "preload": true,
            "rel": "stylesheet",
          },
        ],
      ],
      "scripts": [
        [
          "/static/js/app.client.8dc14fd2c9e52eef.js",
          {
            "async": "",
            "crossorigin": "anonymous",
          },
        ],
        [
          "/static/js/locale/en.3c6a5e7a55bb2ab4.js",
          {
            "async": "",
            "crossorigin": "anonymous",
          },
        ],
        [
          "/static/js/vendors.0e456297851f0a3a.js",
          {
            "async": "",
            "crossorigin": "anonymous",
          },
        ],
      ],
      "styles": [],
    },
  },
  "meta": "",
  "revivalCache": "<script id="ima-revival-cache">(function (root) {
      root.$IMA = root.$IMA || {};
      $IMA.Cache = {};
    })(typeof window !== 'undefined' && window !== null ? window : global);
    </script>",
  "revivalSettings": "<script id="ima-revival-settings">(function (root) {
      root.$Debug = true;
      root.$IMA = root.$IMA || {};
      $IMA.SPA = false;
      $IMA.$PublicPath = "";
      $IMA.$RequestID = "now-0-UUID";
      $IMA.$Language = "en";
      $IMA.$Env = "undefined";
      $IMA.$Debug = true;
      $IMA.$Version = "undefined";
      $IMA.$App = undefined;
      $IMA.$Protocol = "undefined";
      $IMA.$Host = "undefined";
      $IMA.$Path = "undefined";
      $IMA.$Root = "undefined";
      $IMA.$LanguagePartPath = "undefined";
    })(typeof window !== 'undefined' && window !== null ? window : global);
    </script>",
  "runner": "<script id="ima-runner">runner#{scriptResources}</script>",
  "scriptResources": "{\\"scripts\\":[[\\"/static/js/app.client.8dc14fd2c9e52eef.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js/locale/en.3c6a5e7a55bb2ab4.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js/vendors.0e456297851f0a3a.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}]],\\"esScripts\\":[[\\"/static/js.es/app.client.2106a34c6b8bbad8.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js.es/locale/en.371127fdbfbe93d2.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js.es/vendors.a873907f25297544.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}]]}",
  "styles": "<link as="style" href="/static/css/app.d0ad44d05f82db5f.css" rel="preload" type="text/css" /><link href="/static/css/app.d0ad44d05f82db5f.css" rel="stylesheet" />",
}
`;

exports[`responseUtilsFactory processContent should allow overrides through custom $Source definition 1`] = `
"<html><script id="ima-revival-settings">(function (root) {
      root.$Debug = true;
      root.$IMA = root.$IMA || {};
      $IMA.SPA = false;
      $IMA.$PublicPath = "";
      $IMA.$RequestID = "now-0-UUID";
      $IMA.$Language = "en";
      $IMA.$Env = "undefined";
      $IMA.$Debug = true;
      $IMA.$Version = "undefined";
      $IMA.$App = undefined;
      $IMA.$Protocol = "undefined";
      $IMA.$Host = "undefined";
      $IMA.$Path = "undefined";
      $IMA.$Root = "undefined";
      $IMA.$LanguagePartPath = "undefined";
    })(typeof window !== 'undefined' && window !== null ? window : global);
    </script><script id="ima-runner">runner{\\"esScripts\\":[[\\"/static/js/app.client.8dc14fd2c9e52eef.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js/locale/en.3c6a5e7a55bb2ab4.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js/vendors.0e456297851f0a3a.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],\\"custom-script-src\\"]}</script></html>"
`;

exports[`responseUtilsFactory processContent should interpolate revival scripts into page content 1`] = `
"<html><link as="style" href="/static/css/app.d0ad44d05f82db5f.css" rel="preload" type="text/css" /><link href="/static/css/app.d0ad44d05f82db5f.css" rel="stylesheet" /><script id="ima-revival-settings">(function (root) {
      root.$Debug = true;
      root.$IMA = root.$IMA || {};
      $IMA.SPA = false;
      $IMA.$PublicPath = "";
      $IMA.$RequestID = "now-0-UUID";
      $IMA.$Language = "en";
      $IMA.$Env = "undefined";
      $IMA.$Debug = true;
      $IMA.$Version = "undefined";
      $IMA.$App = undefined;
      $IMA.$Protocol = "undefined";
      $IMA.$Host = "undefined";
      $IMA.$Path = "undefined";
      $IMA.$Root = "undefined";
      $IMA.$LanguagePartPath = "undefined";
    })(typeof window !== 'undefined' && window !== null ? window : global);
    </script><script id="ima-runner">runner{\\"scripts\\":[[\\"/static/js/app.client.8dc14fd2c9e52eef.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js/locale/en.3c6a5e7a55bb2ab4.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js/vendors.0e456297851f0a3a.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}]],\\"esScripts\\":[[\\"/static/js.es/app.client.2106a34c6b8bbad8.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js.es/locale/en.371127fdbfbe93d2.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}],[\\"/static/js.es/vendors.a873907f25297544.js\\",{\\"async\\":\\"\\",\\"crossorigin\\":\\"anonymous\\"}]]}</script></html>"
`;
